quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus.test/src/main/java/com/redhat/microprofile/jdt/quarkus/renarde/RenardeJaxRsTest.java

```diff
@@ -0,0 +1,125 @@
+import org.eclipse.lsp4j.SymbolKind;
+import org.eclipse.lsp4mp.commons.MicroProfileJavaCodeLensParams;
+import org.eclipse.lsp4mp.jdt.core.BasePropertiesManagerTest;
+import org.eclipse.lsp4mp.jdt.core.PropertiesManagerForJava;
+import org.eclipse.lsp4mp.jdt.core.utils.IJDTUtils;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTMicroProfileUtils;
+import org.junit.Test;
+
+/**
+ * Tests for the CodeLens features introduced by {@link com.redhat.microprofile.jdt.internal.quarkus.renarde.java.RenardeJaxRsInfoProvider}.
+ */
+public class RenardeJaxRsTest extends BasePropertiesManagerTest {
+
+	private static String quarkus_renarde_todo = "quarkus-renarde-todo";
+
+	@Test
+	public void codeLens() throws Exception {
+		IJavaProject javaProject = loadMavenProject(quarkus_renarde_todo);
+
+		assertNotNull(javaProject);
+
+		MicroProfileJavaCodeLensParams params = new MicroProfileJavaCodeLensParams();
+		params.setCheckServerAvailable(false);
+		IFile javaFile = javaProject.getProject()
+				.getFile(new Path("src/main/java/rest/Application.java"));
+		params.setUri(javaFile.getLocation().toFile().toURI().toString());
+		params.setUrlCodeLensEnabled(true);
+
+		assertCodeLens(params, JDT_UTILS, //
+				ca("http://localhost:8080/", "", r(14, 20, 14, 20)), //
+				ca("http://localhost:8080/about", "", r(19, 25, 19, 25)), //
+				ca("http://localhost:8080/Application/test", "", r(9, 30, 9, 30)));
+	}
+
+	@Test
+	public void workspaceSymbols() throws Exception {
+		IJavaProject javaProject = loadMavenProject(quarkus_renarde_todo);
+
+		assertNotNull(javaProject);
+
+		assertWorkspaceSymbol(javaProject, JDT_UTILS, //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""));
+	}
+
+	private static void assertCodeLens(MicroProfileJavaCodeLensParams params, IJDTUtils utils, CodeLens... expected) throws JavaModelException {
+		List<? extends CodeLens> actual = PropertiesManagerForJava.getInstance().codeLens(params, utils,
+				new NullProgressMonitor());
+		assertEquals(expected.length, actual.size());
+
+		for (int i = 0; i < expected.length; i++) {
+			assertEquals(expected[i], actual.get(i));
+		}
+	}
+
+	private static void assertWorkspaceSymbol(IJavaProject javaProject, IJDTUtils utils, SymbolInformation... expected) throws JavaModelException {
+		List<SymbolInformation> actual = PropertiesManagerForJava.getInstance().workspaceSymbols(JDTMicroProfileUtils.getProjectURI(javaProject), utils, new NullProgressMonitor());
+		assertEquals(expected.length, actual.size());
+
+		for (int i = 0; i < expected.length; i++) {
+			// Only assert that the name is equal
+			assertEquals(expected[i].getName(), actual.get(i).getName());
+			assertEquals(expected[i].getKind(), actual.get(i).getKind());
+		}
+	}
+
+	private static CodeLens ca(String title, String commandId, Range range) {
```

Comment:
MicroProfileForJavaAssert from lsp4mp should already defines those following methods, please use it and define it if they doesn't exist.